php/Client.php

```diff
@@ -198,7 +199,79 @@ public function write(string $type, array $data, array $plain = null): Record
 
         return $this->decrypt_record(Record::decode((string) $resp->getBody()));
     }
-
+    /**
+     * Method for an authorizer to share data it has been authorized to share.
+     *
+     * @param string $writer_id The original writer client id
+     * @param string $reader_id The intended reader of the records
+     * @param string $record_type The record type to share
+     *
+     * 
+     *
+     * @throws \NotFoundException If there is an error while sharing record.
+     */
+    public function share_on_behalf_of(string $writer_id, string $reader_id, string $record_type)
+    {
+        
+        $id = $this->config->client_id;
+        $ak = $this->conn->get_access_key($writer_id, $writer_id, $id, $record_type);
+        try {
+            $this->conn->put_access_key($writer_id, $writer_id, $reader_id, $record_type, $ak);
+        } catch(RequestException $e) {
+            if(!$e->getCode() == 409){
+                throw new NotFoundException("Could not update AK");
+            }
+        }
+        
+        $allow = new \stdClass();
+        $allow->allow = [['read' => new \stdClass()]];
+        $path = $this->conn->uri('v1', 'storage', 'policy',$writer_id, $writer_id, $reader_id, $record_type);
+        $this->conn->put($path, $allow);
+        
+    }
+    /**
+     * Method for an authorizer to unshare data it has been authorized to share.
+     *
+     * @param string $writer_id The original writer client id
+     * @param string $reader_id The intended reader of the records
+     * @param string $record_type The record type to share
+     *
+     * 
+     *
+     * @throws \RuntimeException If there is an error while revoking access.
+     */
+    public function revoke_on_behalf_of(string $writer_id, string $reader_id, string $record_type)
+    {
+        
+        $id = $this->config->client_id;
+        
+        $deny = new \stdClass();
+        $deny->deny = [['read' => new \stdClass()]];
+        $path = $this->conn->uri('v1', 'storage', 'policy',$writer_id, $writer_id, $reader_id, $record_type);
+        $this->conn->put($path, $deny);
+        $this->conn->delete_access_key($writer_id, $writer_id, $reader_id, $record_type);
+        
+    }
+    /**
+     * Method to get a list of all clients and record types that this client is 
+     * allowed to act as an authorizer on.
+     *
+     *
+     * @return Array
+     *
+     * @throws \NotFoundException If there is an error while fetching the data.
+     */
+    public function get_authorized_by()
+    {
+        $path = $this->conn->uri('v1', 'storage', 'policy','granted');
+        try {
+            $resp = $this->conn->get($path);
+        } catch (RequestException $re) {
+            throw new NotFoundException("Could not retrieve data");
+        }
+        return json_decode($resp->getBody()->getContents());
```

Comment:
Does the php SDK do anything with types, or is it just throwing decoded json bodies around?

Reply:
It is doing types:
https://github.com/tozny/e3db-php/tree/master/php/Types

I agree for consistency this should return a new type rather than a `stdClass` instance.